content/en/docs/how-tos/adding-a-new-secret-to-ci.md

```diff
@@ -4,9 +4,66 @@ date: 2020-12-21T10:08:01-04:00
+
+## Add a new secret
+
 Choose a new namespace, and create a pull request to add a folder with the same name in [release/core-services](https://github.com/openshift/release/tree/master/core-services). The folder should contain the manifests for the namespace and the RBACs regarding to the namespace. If the desired namespace exists already, ask the owners of the namespace if it can be shared. After the pull request is merged, the manifests in the folder will be applied automatically to `api.ci`.
 
 Log into `api.ci` and create the secret in the namespace chosen above.
 
 Use the secret as the source in [the mapping file](https://github.com/openshift/release/tree/master/core-services/secret-mirroring).
 The secret and any modification on it afterwards will be populated to the targeting namespace on [all clusters](/docs/getting-started/useful-links/#clusters) in the build farm.
+
+```yaml
+- from:
+    namespace: some-namespace
+    name: source-some-secret
+  to:
+    namespace: test-credentials # the suggested namespace holding all secrets used in a step
+    name: target-some-secret # ensure the secret name unique cross all teams in our organization
+```
+
+## Use a secret in a job
+
+* The most common case is to use secrets in a [step](/docs/architecture/step-registry/#step). In this case, we suggest the user to mirror secrets to `test-credentials` namespace. The pod which runs the step can access the secrets defined in `credentials` stanza of the step definition. see [injecting-custom-credentials](https://docs.ci.openshift.org/docs/architecture/step-registry/#injecting-custom-credentials) for details.
+
+* For a job which is generated from `ci-operator`'s config and does not use steps, we have to use `ci` as the targeting namespace in secret-mirroring. And then mount the secrets via `secrets` stanza in the ci-operator's config, e.g.,
+
+```yaml
+tests:
+- as: "vet"                  # names this test "vet"
+  commands: "go vet ./..."   # declares which commands to run
+  container:
+    from: "src"              # runs the commands in "pipeline:src"
+  secrets:
+  - mount_path: /secret      # mount path of the extracted files from the secret
+    name: secret-name-in-ci  # the secret name in the ci namespace
+```
+
+* For a job which does not even use `ci-operator` at all, i.e. [handcrafted jobs](/docs/how-tos/contributing-openshift-release/#handcrafted-jobs), `ci` is the targeting namespace in secret-mirroring. As stated there, **creating handcrafted jobs is discouraged**. The follow example shows how to use secrets in a job definition:
```

Comment:
We should not suggest users to write hand-crafted jobs so this should be omitted.

Reply:
Well we don't suggest it, rather the opposite. The fact is that some teams have handcrafted jobs and we'll have them in the future - I'd much rather have a doc to send to the people that need it than explain them on Slack. Fringe cases unfortunately need docs, too.